src/test/java/seedu/address/logic/commands/AddCommandTest.java

```diff
@@ -163,6 +163,11 @@ public void redoTaskManager() {
         public void commitTaskManager() {
             throw new AssertionError("This method should not be called.");
         }
+
```

Comment:
Undo/redo won't work if your statePointer is pointing at 0.

Reply:
Phone call discussion summary:
Problem:
Some commands have multiple edits to a mutable data structure (unique task list) **in a single execution.** How do we ensure that there is reversibility after a partial commit.
Two possible implementations. 
1) Ensure that all checks in the execute function of the command are comprehensive. So that there is no likelihood of exception thrown after partial commit. Benefit is that this implementation reduces complexity.
2) Implement a partial rollback function. (seen here). Benefit is that it ensures that it is possible to rollback after partial commit.